src/lib/one_or_two/one_or_two.ml

```diff
@@ -0,0 +1,92 @@
+open Core
+open Async
+
+module Stable = struct
+  module V1 = struct
+    module T = struct
+      type 'a t = [`One of 'a | `Two of 'a * 'a]
+      [@@deriving bin_io, equal, compare, hash, sexp, version, yojson]
+    end
+
+    include T
+  end
+
+  module Latest = V1
+end
+
+type 'a t = 'a Stable.V1.t [@@deriving compare, equal, hash, sexp, yojson]
+
+let length = function `One _ -> 1 | `Two _ -> 2
+
+let to_list = function `One a -> [a] | `Two (a, b) -> [a; b]
+
+let group_sequence : 'a Sequence.t -> 'a t Sequence.t =
```

Comment:
I'm not of a fan of using `'` to distinguish variables, just saying

Reply:
I might argue with you in another situation but once you reach the level of triple prime I agree it's bad.

Reply:
This is all bikeshedding but I like the `'` up to two, it's a nice way to say it's a changed version of some variable.